File: Programmers/[12899] 124 나라의 숫자/SUbbb/12899.java
class Solution {
    public String solution(int n) {
        StringBuilder sb = new StringBuilder();
        
        while (n > 0) {
            int remain = n % 3;
            
            if (remain == 0) remain = 4;
            sb.append(remain);
            
            n = (n - 1) / 3;
        }
        
        return sb.reverse().toString();
    }
}

File: Programmers/[12899] 124 나라의 숫자/KingDonggyu/12899.js
function solution(n) {
  const map124 = [4, 1, 2];
  const answer = [];

  while (n) {
    answer.push(map124[n % 3]);
    n = Math.floor((n - 1) / 3);
  }

  return answer.reverse().join('');
}
